packages/blade/src/components/Table/_decisions/decisions.md

```diff
@@ -0,0 +1,375 @@
+            <Button>Export</Button>
+        </ToolbarActions>
+    </TableToolbar>
+
+    <TableHeader isSticky={true}>
+      <TableHeaderRow>
+         <TableHeaderCell headerKey='firstName'>
+          First Name
+        </TableHeaderCell>
+        <TableHeaderCell headerKey='lastName'>
+          Last Name
+        </TableHeaderCell>
+        <TableHeaderCell headerKey='balance'>
+          Balance
+          <Tooltip content='Current bank balance'>
+            <InfoIcon size='medium' />
+          </Tooltip>
+        </TableHeaderCell>
+       </TableHeaderRow>
+     </TableHeader>
+
+    <TableBody>
+      <TableRow onClick={console.log} isDisabled={false}>
+        <TableCell>
+          {tableData.firstName}
+        </TableCell>
+        <TableCell>
+          {tableData.lastName}
+        </TableCell>
+        <TableCell>
+          {tableData.balance}
+          <Badge>Low</Badge>
+        </TableCell>
+      <TableRow>
+    </TableBody>
+
+    <TableFooter>
+      <TableFooterRow>
+        <TableFooterCell columnStart='1' columnEnd='4'>
+          Total
+        </TableFooterCell>
+        <TableFooterCell>
+          {computedTotalBalance}
+        </TableFooterCell>
+      </TableFooterRow>
+    </TableFooter>
+
+    <TablePagination 
+      navigationType='compact' 
+      label='1-10 of 100'
+      labelPosition='left'
+      showRowCountPicker
+      rowCount='25'
+      onPaginationChange={console.log}
+      onRowCountPickerChange={console.log}
+      /> 
+   </>
+  )
+}
+</Table>
+```
+
+##### Pros
+- Composable API is more intuitive and resembles the native HTML structure of a table
+- It is easier to understand since the API design is similar to rest of the Blade components
+- It is easier to extend and add new feature to individual table components
+- We could leverage tree-shaking to only import the components that are being used in the consumer's table
+
+##### Cons
+- Composable API is more verbose and requires more effort to create a table by composing together multiple components
+- Most popular table libraries like AG Grid, React Table, etc. use a compact API which is battle tested and developers are more familiar with the same
+
+#### 2. Compact API
+Compact API is a pattern where we have a single component that takes in all the data and renders a table. This approach is more concise and requires less effort to create a table. However, it is also less intuitive and does not resemble the native HTML structure of a table.
+
+```jsx
+// Mock data & functions
+const data = [{
+    firstName: 'John',
+    lastName: 'Doe',
+    balance: 1000
+},
+{
+    firstName: 'Jane',
+    lastName: 'Doe',
+    balance: 2000
+}];
+
+
+const sortFunctions = {
+  firstName: (array) => array.sort((a, b) => a.name.localeCompare(b.name)),
+};
+
+const onSelectionChange = (selectedItems) => {
+  console.log(selectedItems);
+};
+
+const columnDefinition = [
+    {
+        header: 'First Name',
+        headerKey: 'firstName',
```

Comment:
what is headerKey required for?

Reply:
Required for sorting internally. But this can be something like `key` or `id`. Some unique identifier